applications/skyekiwi-protocol.md

```diff
@@ -0,0 +1,200 @@
+
+### Overview
+
+* **Total Estimated Duration:** 4 months (16 weeks)
+* **Full-Time Equivalent (FTE):**  2 **FTE**
+* **Total Costs:** $30,000 (payable in DAI)
+
+### Milestone 1 - Phase 1
+
+**Phase One** - PoC Phase: Connects to a smart contract of a Substrate style WASM smart contract enabled blockchain; the Crust Network for storage; Public IPFS Pin Service & Gateways, with a SkyeKiwi ran IPFS Cluster as backup IPFS nodes. 
+
+* **Estimated duration:** 2 month
+* **FTE:**  2
+* **Costs:** 12,000USD
+
+| Number | Deliverable | Specification |
+| -----: | ----------- | ------------- |
+| 0a. | License | Apache 2.0 |
+| 0b. | Documentation | We will provide both **inline documentation** of the code and a basic **tutorial** that explains how a user can integrate the SkyeKiwi Protocol into their project. |
+| 0c. | Testing Guide | Core functions will be fully covered by unit tests to ensure functionality and robustness. In the guide, we will describe how to run these tests on both Node.js and Browser environment. |
+| 1. | Core Protocol | Implements the `upstream` `downstream` `updateEncryptionSchema` and `verifyProofOfAccess` APIs. |
+| 2. | Smart Contract | Implements and test for the smart contract used for registering secrets and access control write access to the secret. |
+| 3. | Polkadot.js | Add in encryption/decryption functionality to `@polkadot/keyring` and `@polkadot/extension` so that the protocol can run without the needs to read the private key of users. |
+### Milestone 2 - Phase 2
+
+**Phase Two**: Connects to the SkyeKiwi Network `pallet-secrets` instead of a smart contract for general secrets registry; the Crust Network for storage; Public IPFS Pin Service & temporary IPFS nodes from the Crust Network ecosystem solutions. 
+
+**Note:** We won't be 100% sure of the availability of the Crust Network temporary IPFS solution but will actively engage with the Crust Network for this part. Therefore, we'd like to mark it as optional. Worst thing worst, we still have our own IPFS cluster running. 
+
+* **Estimated duration:** 2 month
+* **FTE:**  2
+* **Costs:** 18,000USD
+
+| Number | Deliverable         | Specification                                                |
+| -----: | ------------------- | ------------------------------------------------------------ |
+|    0a. | License             | Apache 2.0                                                   |
+|    0b. | Documentation       | We will provide both **inline documentation** of the code and a basic **tutorial** that explains how a user can integrate the SkyeKiwi Protocol into their project. |
+|    0c. | Testing Guide       | Core functions will be fully covered by unit tests to ensure functionality and robustness. In the guide, we will describe how to run these tests on both Node.js, Browser and Rust environment. |
+|     1. | `pallet-secrets`    | Port the `ink!` smart contract to be a runtime pallet and integrate it to the SkyeKiwi Protocol source code. |
```

Comment:
What's the purpose of implementing this twice, why don't you leave out the smart contract?

Reply:
I was only worried this would mean you'll be doing some of the work twice, but those are very good reasons to intentionally do so.